### PR TITLE
feat(react-components): Add ground plane rotation 0.52.0

### PR DESCRIPTION
**Pull request by greenbech**
#### Type of change

Feat  

#### Jira ticket :blue_book:

No JIRA ticket.

## Description :pencil:

This PR makes the ground plane load the Euler rotation and scale from DM. Before it only loaded the position.

## How has this been tested? :mag:

This has been tested by `yalc`-ing with fusion.

## Test instructions :information_source:

Currently, I don't think there is any other easy way of testing than `yalc`-ing this into fusion and see that the rotation is the same as in SceneBuilder.

## Checklist :ballot_box_with_check:

- [ ] I am proud of this feature.
- [ ] I have performed a self-review of my own code.
- [ ] I have added PR type (Feat, Bug, Chore, Test, Docs, Style, Refactor) to the PR title.
- [ ] I have manually tested this for different scenarios (different models, formats, devices, browsers).
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the public documentation.
- [ ] I have added unit and visuals tests to prove that my feature works to the best of my ability.
- [ ] I have refactored the code for readability to the best of my ability.
- [ ] I have checked that my changes do not introduce regressions in the public documentation.
- [ ] I have outlined any known defects / lacking capabilities in the contents of this PR.
- [ ] I have listed any remaining work that should follow this PR in the description or jira/miro/etc.
- [ ] I have added TSDoc to any public facing changes.
- [ ] I have added "developer documentation" in any package README.md that is related / applicable to this PR.
- [ ] I have noted down and am currently tracking any technical debt introduced in this PR.
- [ ] I have thoroughly thought about the architecture of this implementation.
- [ ] I have asked peers to test this feature.

### CODE REVIEW

File: react-components/src/hooks/useGroundPlaneFromScene.tsx [hunk 1]
```diff
@@ -88,14 +89,10 @@ export const useGroundPlaneFromScene = (sceneExternalId: string, sceneSpaceId: s
...
       const mesh = new Mesh(geometry, material);
-      mesh.position.set(
-        groundPlane.translationX,
-        groundPlane.translationY,
-        groundPlane.translationZ
-      );
-      mesh.rotation.set(-Math.PI / 2, 0, 0);
 
-      mesh.position.applyMatrix4(CDF_TO_VIEWER_TRANSFORMATION);
+      const matrix4 = transformation3dToMatrix4(groundPlane).multiply(CDF_TO_VIEWER_TRANSFORMATION);
```
Comment (pramodcog) [inline react-components/src/hooks/useGroundPlaneFromScene.tsx:93]: Cant we just convert Euler rotation to quaternion and apply to the mesh?

Comment (greenbech) [reply react-components/src/hooks/useGroundPlaneFromScene.tsx:93]: We should also apply the scale. Isn't this what I'm doing? I am open to changing the syntax — do you have a concrete suggestion?

Comment (pramodcog) [reply react-components/src/hooks/useGroundPlaneFromScene.tsx:93]:
```
   // Apply position from groundPlane
    mesh.position.set(groundPlane.position.x, groundPlane.position.y, groundPlane.position.z);

    // Convert Euler rotation to quaternion and apply to the mesh
    const euler = new Euler(groundPlane.rotation.x, groundPlane.rotation.y, groundPlane.rotation.z, groundPlane.rotation.order ?? 'XYZ');
    mesh.quaternion.setFromEuler(euler);

    // Apply scaling from groundPlane
    mesh.scale.set(groundPlane.scaleX, groundPlane.scaleY, groundPlane.scaleZ);
```
I feel this is much simpler, unless it causes an issue

Comment (greenbech) [reply react-components/src/hooks/useGroundPlaneFromScene.tsx:93]: After chatting with @haakonflatval-cognite, I think maybe the simplest way of doing this is what I updated it to now:

```typescript
      mesh.matrix.copy(matrix4);
      mesh.matrixAutoUpdate = false;
```

Do you agree?

**Review by haakonflatval-cognite — APPROVED** [review]
LGTM :+1:

**Review by anders-hopland — APPROVED** [review]
LGTM

**Pull request merged**